### PR TITLE
Changes permission of libvirt keyring

### PR DESCRIPTION
**Pull request by nilamdyuti**
Fixes: #1445

Signed-off-by: Nilamdyuti Goswami <ngoswami@redhat.com>

### CODE REVIEW

File: roles/ceph-mon/tasks/ceph_keys.yml [hunk 1]
```diff
@@ -99,13 +99,32 @@
     - "{{ ceph_keys.get('stdout_lines') | default([]) }}"
   when: cephx
 
+- name: check if libvirt keyring exists
+  command: find /etc/ceph -name "*.libvirt.keyring"
```
Comment (leseb) [inline roles/ceph-mon/tasks/ceph_keys.yml]: will the `*` work with the `command` module?

File: roles/ceph-mon/tasks/ceph_keys.yml [hunk 2]
```diff
@@ -99,13 +99,32 @@
...
+  always_run: true
+  run_once: true
+  when: cephx
+
+- name: set ceph libvirt keyring permission
+  file:
+    path: "/etc/ceph/{{ cluster }}.client.libvirt.keyring"
+    owner: "ceph"
+    group: "ceph"
+    mode: "0640"
```
Comment (leseb) [inline roles/ceph-mon/tasks/ceph_keys.yml]: Don't you want these owner/group/mode to be configurable?

File: roles/ceph-mon/tasks/ceph_keys.yml (hunk 1 above)
Comment (nilamdyuti) [inline roles/ceph-mon/tasks/ceph_keys.yml]: @leseb Yes, you are right. Wildcard ideally wouldn't work with command module. I have to use the shell module for the command, take the output in a register variable, then use command module. Then it should work.

File: roles/ceph-mon/tasks/ceph_keys.yml (hunk 2 above)
Comment (nilamdyuti) [inline roles/ceph-mon/tasks/ceph_keys.yml]: @leseb Sorry, I missed that. I will make the changes.

**Comment by nilamdyuti** [comment]
@leseb Made the changes. Kindly review.

**Comment by nilamdyuti** [comment]
@leseb Are the changes acceptable or modification is required?

**Comment by leseb** [comment]
@nilamdyuti I don't see anything.

**Comment by nilamdyuti** [comment]
@leseb Sorry, my bad. I had amended the commit. Maybe that's why you don't see the diff. The new changes can be seen in between line 102 to line 122. Old unchanged changes are in line 86 and 130.

**Comment by nilamdyuti** [comment]
@leseb Were you able to see the changes?

**Comment by leseb** [comment]
@nilamdyuti nop, can you?

**Comment by nilamdyuti** [comment]
@leseb That's strange. Yes, I can. As I said before, the changes can be seen from line 102 to line 122.

**Comment by nilamdyuti** [comment]
@leseb I can see the changes in both `unified` and `split` view under `Files changed` tab.

File: roles/ceph-mon/tasks/ceph_keys.yml [hunk 3]
```diff
@@ -99,13 +99,35 @@
...
+    command: chown {{ owner }}:{{ group }} {{ path }} ; chmod {{ mode }} {{ path }}
```
Comment (leseb) [inline roles/ceph-mon/tasks/ceph_keys.yml]: There is something wrong with your patch here, you included the 'command' module in the file module.
Also, where are those variables declared? owner? group? path? mode?

Comment (nilamdyuti) [inline roles/ceph-mon/tasks/ceph_keys.yml]: @leseb My bad. Making the changes.

**Comment by nilamdyuti** [comment]
@leseb Committed the changes. Kindly review.

File: roles/ceph-mon/tasks/ceph_keys.yml [hunk 4]
```diff
@@ -99,13 +99,32 @@
...
+- name: check if libvirt keyring exists
+  command: find /etc/ceph -name "{{ cluster }}.client.libvirt.keyring"
+  changed_when: false
+  register: is_libvirt_keyring_exists
+  always_run: true
+  run_once: true
+  when: cephx
+
+- name: set ceph libvirt keyring permission
+  shell: chown ceph:ceph *.libvirt.keyring ; chmod 0640 *.libvirt.keyring
```
Comment (leseb) [inline roles/ceph-mon/tasks/ceph_keys.yml]: Previously you were using the `file` module which is exactly what you should be doing.
My last comment was the owner/mode might need to be configurable with variables so this is not hardcoded.

Comment (nilamdyuti) [inline roles/ceph-mon/tasks/ceph_keys.yml]: @leseb Okay. Making the changes.

Comment (nilamdyuti) [inline roles/ceph-mon/tasks/ceph_keys.yml]: @leseb In task `set keys permissions` of this file `roles/ceph-mon/tasks/ceph_keys.yml`  owner/group/mode are not configurable with variables as well. They are simply used the way they are used in a `file` module. That's what I had done in my first commit for the task `set ceph libvirt keyring permission` that I had added.

Do owner/group/mode actually need to be configurable with variables?

Comment (nilamdyuti) [inline roles/ceph-mon/tasks/ceph_keys.yml]: @leseb Added the `file` module again. Kindly check.

Comment (nilamdyuti) [inline roles/ceph-mon/tasks/ceph_keys.yml]: @leseb Are the changes acceptable or modification is required?

**Comment by nilamdyuti** [comment]
@leseb Can you please check if the changes are acceptable?

**Comment by leseb** [comment]
@nilamdyuti I don't think this is the right approach to fix this. I don't really want to end up in a situation where everyone adds new code for a new key. Let me think about this.

**Comment by guits** [comment]
This PR has been opened for more than a year, if you think it's still worth to pursue feel free to re-open.

**Pull request closed without merging**